### PR TITLE
docs: improve data fetching section

### PR DESCRIPTION
**Pull request by DarkGhostHunter**
### 🔗 Linked issue

None

### ❓ Type of change

- [X] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Minor clarifications to data fetching. There are concepts that are introduced before the previous completes introduction, so I added a more clearer presentation of these concepts in the beggining.

Also, added the section for `immediate: false`.

Full details on the diff.

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [X] I have updated the documentation accordingly.

### CODE REVIEW

File: docs/1.getting-started/6.data-fetching.md [hunk 1]
```diff
@@ -185,6 +219,10 @@ const { data: mountains } = await useFetch('/api/mountains', {
 })
 ```
 
+::alert{type="warning"}
+Both `pick` and `transform` don't prevent the unwanted data to be transmitted. They only work over the data already received.
```
Comment (danielroe) [inline docs/1.getting-started/6.data-fetching.md:223]:
```suggestion
Both `pick` and `transform` don't prevent the unwanted data from being fetched initially. But they will prevent unwanted data from being added to the payload transferred from server to client.
```

File: docs/1.getting-started/6.data-fetching.md [hunk 2]
```diff
@@ -215,23 +253,58 @@ const { data, error, execute, refresh } = await useFetch('/api/users')
...
+  </div>
+</template>
 ```
 
+For finer control, the `status` variable can be:
+
+* `idle` when the fetch hasn't started,
+* `pending` when a fetch has started but not yet completed,
+* `error` when the fetch fails,
+* `success` when the fetch is completed successfully.
```
Comment (danielroe) [inline docs/1.getting-started/6.data-fetching.md:306]:
```suggestion
* `idle` when the fetch hasn't started
* `pending` when a fetch has started but not yet completed
* `error` when the fetch fails
* `success` when the fetch is completed successfully
```

File: docs/1.getting-started/6.data-fetching.md [hunk 3]
```diff
@@ -71,34 +83,53 @@ The `ofetch` library is built on top of the `fetch` API and adds handy features
...
+  /* Wait for all fetches to complete */
+  Promise.all([coupons, offers])
+
+  const 
+
+  /* All set, return them */
+  return {
+    coupons: coupons.data,
+    offers: offers.data
+  }
```
Comment (danielroe) [inline docs/1.getting-started/6.data-fetching.md:130]:
```suggestion
  const [coupons, offers] = await Promise.all([$fetch('/cart/coupons'), $fetch('/cart/offers')])

  return {
    coupons,
    offers
  }
```

**Review by danielroe — COMMENTED** [review]
These are nice improvements - thank you ❤️

**Comment by DarkGhostHunter** [comment]
All reviews approved.

**Review by danielroe — APPROVED** [review]
Nice - thank you ❤️

**Pull request merged**